### PR TITLE
fix: Unable to set url prefix for the application

### PR DESCRIPTION
**Pull request by wasd845**
---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#238: Unable to set url prefix for the application](https://issuehunt.io/repos/271918885/issues/238)
---
</details>

### CODE REVIEW

File: packages/router/src/index.ts [hunk 1]
```diff
@@ -127,23 +130,28 @@ export const pushMiddleware = <Req extends any = any, Res extends any = any>(mw:
...
-  type
+  type,
+  fullPaths
 }: MethodHandler<Req, Res> & {
   method?: Method
   handlers?: RouterHandler<Req, Res>[]
+  fullPaths?: string[]
 }) => {
-  const m = createMiddlewareFromRoute<Req, Res>({ path, handler, method, type })
+  const m = createMiddlewareFromRoute<Req, Res>({ path, handler, method, type, fullPath : fullPaths == null ? null : fullPaths[0] })
```
Comment (talentlessguy) [inline packages/router/src/index.ts:140]: could you please replace it with `fullPaths?.[0]`?

**Review by talentlessguy — CHANGES_REQUESTED** [review]

**Comment by talentlessguy** [comment]
woah! the idea with `fullPaths` is actually smart

could you also add tests please?

**Comment by wasd845** [comment]
> woah! the idea with `fullPaths` is actually smart
> 
> could you also add tests please?

how to add tests in Pull Request?

**Comment by talentlessguy** [comment]
@wasd845 just push them to the PR from your machine

it will automatically append here

**Comment by wasd845** [comment]
> woah! the idea with `fullPaths` is actually smart
> 
> could you also add tests please?

I have added the "set url prefix for the application" test, Can it work automatically? Or do I need to do other operations?

**Comment by talentlessguy** [comment]
> > woah! the idea with `fullPaths` is actually smart
> > could you also add tests please?
> 
> I have added the "set url prefix for the application" test, Can it work automatically? Or do I need to do other operations?

please wait until CI ends running

**Comment by talentlessguy** [comment]
@wasd845 you forgot to format your code

run `pnpm format` please

**Comment by talentlessguy** [comment]
@wasd845 tests are failing for some reason :thinking:

**Comment by wasd845** [comment]
> @wasd845 tests are failing for some reason 🤔

Sorry, It seems I need to do more test in my machine locally.

**Pull request closed without merging**